nzshm_common/location/code_location.py

```diff
@@ -13,14 +13,22 @@ class CodedLocation:
     lon: float = field(hash=True)
     resolution: float = field(hash=True)
 
-    def __init__(self, lat, lon, resolution):
+    def __init__(self, lat: float, lon: float, resolution: float) -> None:
+        """
+        Create a CodedLocation instantce.
+
+        Arguments:
+            lat: latitude
+            lon: longitude
+            resolution: the resolution used to resolve the location
+        """
         assert 0 < resolution < 180
 
         self.grid_res = decimal.Decimal(str(resolution).rstrip("0"))
-        self.display_places = max(abs(self.grid_res.as_tuple().exponent), 1)
+        self.display_places = max(abs(self.grid_res.as_tuple().exponent), 1)  # type: ignore
 
         div_res = 1 / float(self.grid_res)
-        places = abs(decimal.Decimal(div_res).as_tuple().exponent)
+        places = abs(decimal.Decimal(div_res).as_tuple().exponent)  # type: ignore
```

Comment:
You could potentially fix these type issues by explicitly wrapping `.as_tuple().exponent` values in an `int()`.

Decimal has some edge case values like "Infinity" and "NaN" that are valid for the type, which gives you exponent values that aren't ints ('F' and 'n' respectively). If you do an explicit int conversion then the edge cases will error out accordingly:

```
ValueError: invalid literal for int() with base 10: 'F'
```

It is still safe to ignore those cases since we've already clamped it to a (0-180) value range with the assertion above, though. An explanatory comment might suffice.

Reply:
Decision: will cast these under another ticket.